### ISSUE TITLE
Commands provided by lines or template should be treated the same for network modules

### ISSUE THREAD

**Issue by jmighion** [ansible/ansible#23502]
##### ISSUE TYPE

 - Feature Idea

##### OS / ENVIRONMENT

N/A

##### SUMMARY

This is similar to #19884 and #22725 but more specific to treating the template exactly the same as provided lines.
Since `lines:` and `src:` are mutually exclusive, I assumed they would be treated the same with regards to the other parameters available in a module, but this is not the case. Commands provided from `lines:` can benefit from the other module parameters like `before:`, `after:`, `parents:`, `match:`, and `replace:`. This makes `lines:` much more robust. The commands from the template are interpolated and matched line by line, but without any added benefits from the other module parameters which leaves it treated like `lines:` set with `match: line` and `replace: line` and no ability to use or edit any other options. If the template was interpolated and put into the same variable that populates `lines:` then templates would have the same flexibility.

##### STEPS TO REPRODUCE

```yaml
- name: dev
  hosts: all
  gather_facts: false
  connection: local
tasks:
    - name: templated
      ios_config:
        src: test.j2
        before: do show version
        match: exact
        parent: ip access-list standard test-group
        replace: block
        provider: "{{ cli }}"
```

test.j2:
```
permit ip.ip.ip.ip 0.0.1.255
permit ip.ip.ip.ip 0.0.31.255
permit ip.ip.ip.ip 0.0.31.255
```

##### EXPECTED RESULTS

Expected to check those lines against the parent, match exact, replace the whole block, and run the before command.

##### ACTUAL RESULTS

```

```

**Comment by gundalow** [comment]
@jmighion Thanks for this. We will have a think about it and get back to you

**Comment by calfonso** [comment]
Hi!

We believe recent commits should resolve this question or problem for you and available with the latest release (2.4)

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:

https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code
Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
